### DISCUSSION TITLE
Fellow Csla Developers and Javascript

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#1629] [General]
Hi All,
So we found out we can't work with DOM from Blazor (I don't know if Microsoft is planning to add DOM manipulation in Blazor). We don't have real complex requirements, one for instance is to 'focus' on certain elements based on business logic. I was wondering what do fellow Csla developers use for 'client' side development? One of our team members worked with React and liked it. We have also used Jquery in the past.
Apparently we can't escape Javascript in the foreseeable future so we might as well get used to it lol.
Kind regards

**Comment by rockfordlhotka** [comment]
Right now [WebAssembly](https://webassembly.org) doesn't support direct DOM interaction. This may happen in the future, but doesn't appear to be a high priority for the wasm team.

Microsoft and Blazor can't do anything beyond what wasm supports, so unless the wasm spec someday includes direct DOM interaction that's not going to happen.

mono-wasm does have interop with JavaScript, as I'm sure you've discovered. So you can have js call C# and C# call js. This is how the entire Blazor UI frame work is implemented.

Also, you should be aware that there are tons of people (commercial products and OSS) building Blazor components. It may be that something you need already exists out there. Or not, in which case you'd have to figure out how to build it - or perhaps work with an existing OSS product or a component vendor you already work with to work with you to build it.

**Reply by JacoJordaan** [reply]
I know the Blazor team is actively working to expand the functionality to do most in c#.
I am sure you should be able to use JSInteropt to do things like this. Maybe not.
Maybe this [link](https://stackoverflow.com/questions/59137973/how-to-set-the-focus-to-an-inputtext-element/59143132#59143132)  can help
